File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.java
class Solution {
    public boolean halvesAreAlike(String s) {
        int count = check(s.toLowerCase(), 0, s.length()/2);
        int count2 = check(s.toLowerCase(),s.length()/2,s.length());
        
        return count == count2;
    }
    
    private static int check(String s, int start, int end){
        int count = 0;
        for(int i=start;i<end;i++){
            if(s.charAt(i) == 'a' || s.charAt(i) == 'e' || s.charAt(i) == 'i' || s.charAt(i) == 'o' || s.charAt(i) == 'u'){
                count++;
            }
        }
        return count;
    }
}

File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.js
/**
 * @param {string} s
 * @return {boolean}
 */
var halvesAreAlike = function(s) {
    let count1 = check(s,0,Math.floor(s.length/2));
    let count2 = check(s,Math.floor(s.length/2), s.length);
    
    return count1 === count2;
    
    function check(s, start, end){
        let count = 0;
        for(let i=start;i<end;i++){
            let ch = s.charAt(i);
            if(ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u' ||ch == 'A' || ch == 'E' || ch == 'I' || ch == 'O' || ch == 'U' ){
                count++;
            }
        }
        return count;
    }
};
